### DISCUSSION TITLE
Linux ARM64 VM Taking Too Much Space Very Quickly & Want to Reduce this Amount of Space or Fix Issue...

### DISCUSSION THREAD

**Discussion by ocowala** [utmapp/UTM#3607] [UTM Help (macOS)]
**Describe the issue**
Hello fellow developers,
My Linux arm64 vm is taking too much space and it is getting out of hand for me. For this VM i put 30 GB Space in the VM and after 1 week of using it, the amount of space left is 2 GB. The only applications I have downloaded is VSCode (idk if you have to download chromium), and g++ compiler. Please help, it would be much appreciated!!! 
**Configuration**
* UTM Version: Latest version
* OS Version: MacOS Monterey
* Apple Silicon M1

**Crash log**
If the app crashed, you need a crash log. To get your crash log, open Console.app, go to `Crash Reports`, and find the latest entry for either QEMU, QEMUHelper, or qemu-\*. Right click and choose `Reveal in Finder`. Attach the report here.

**Debug log**
For all issues, _including_ crashes, you should attach a debug log. Open UTM, and open the settings for the VM you wish to launch. Near the top of the `QEMU` page is `Debug Log`. Turn it on and save the VM. After you experience the issue, open the VM settings again and select `Export Log...` and attach it here.

**Upload VM**
(Optional) If possible, upload the `config.plist` inside the `.utm`. If you do not have this, you can upload the entire `.utm` but note this contains your personal data. Since Github has an attachment size limit, you may want to upload to another service such as Google Drive. Link it here.

**Comment by adespoton** [comment]
Since you're on macOS, try installing qemu (via brew or MacPorts) and using qemu-img to "convert" your disk images to qcow2.  This will shrink the file size down to only the used parts again.  Using the compression flag will shrink it even further, but compressed qcow2 images can't be snapshotted if you use that feature.

There are some other existing threads on here regarding other ways of trimming free space without having to rebuild your disk images, and there's features planned for UTM 3.1 that should do some of this automatically.

**Comment by ocowala** [comment]
I have installed it, using brew, now what do i do?

**Comment by adespoton** [comment]
Here's the script I use:

```
#!/bin/bash
if [ -z "$1" ]
then
echo "Zero out space on guest system using:"
echo "diskutil secureErase freespace 0 /"
echo "or on systems that don't have secureErase,"
echo "dd if=/dev/zero bs=4196 of=tempzero; rm tempzero"
echo "On drive images that can use TRIM, zeroing is not required."
echo "-"
echo "Drop a disk image on here to convert to a compressed QCOW2 image."
echo "Compressed image will have .qcow2 appended and be written to the source folder."
echo "-"
echo "NOTE: compressed qcow2 images CANNOT be used for snapshotting."
echo "      Convert without the -c flag if snapshotting is desired."
echo "-"
else
/opt/homebrew/bin/qemu-img convert -c -O qcow2 "$1" "$1.qcow2"
echo "$1 Done!"
```

If you want VM snapshots, remove the -c option.

Run this script on an image in your .utmconfig/Images/ folder when UTM is not running.  Then delete the original file and rename the .qcow2.qcow2 file to only end with .qcow2.

**Comment by adespoton** [comment]
and the basic way to use it is to run qemu-img convert -O qcow2 <drag inputfile here> <drag input file here again, with .qcow2 on the end>

**Comment by adespoton** [comment]
You'll get more compression if you TRIM or zero out freespace inside the VM before compressing the image file.
